next-frontend/src/components/live/ConnectionPulse.tsx

```diff
@@ -0,0 +1,31 @@
+"use client";
+
+import {
+  CONNECTION_COLORS,
+  ConnectionState,
+} from "@/lib/hooks/useResultsSubscription";
+import { Status } from "@chakra-ui/react";
+import { useT } from "@/lib/i18n/useI18n";
+
+export default function ConnectionPulse({
+  connectionState,
+}: {
+  connectionState: ConnectionState;
+  animationDuration?: number;
+}) {
+  const { t } = useT();
+
+  const connectionColor = CONNECTION_COLORS[connectionState];
+
+  return (
+    <Status.Root colorPalette={connectionColor}>
+      <Status.Indicator
+        animationName="ping"
+        animationDuration="1s"
+        animationTimingFunction="cubic-bezier(0, 0, 0.2, 1)"
+        animationIterationCount="infinite"
+      />
```

Comment:
Cool! How do you stop the animation though based on the current `connectionState`?

In the video that you posted the dot is indeed changing color (and text), but it seems to be always blinking. I think only a (slow) blink when it's actually connected and live (green) makes the most sense.